tsfile/src/main/java/org/apache/iotdb/tsfile/utils/BytesUtils.java

```diff
@@ -495,19 +518,35 @@ public static boolean bytesToBool(byte[] b, int offset) {
    * @param width bit-width
    */
   public static void longToBytes(long srcNum, byte[] result, int pos, int width) {
-    int temp = 0;
-    for (int i = 0; i < width; i++) {
-      temp = (pos + width - 1 - i) / 8;
-      try {
-        result[temp] = setByteN(result[temp], pos + width - 1 - i, getLongN(srcNum, i));
-      } catch (Exception e) {
-        LOG.error(
-            "tsfile-common BytesUtils: cannot convert a long {} to a byte array, pos {}, width {}",
-            srcNum,
-            pos,
-            width,
-            e);
+    if (width < 64) {
+      srcNum = srcNum & ~(-1L << width);
```

Comment:
We can consider to remove this `pre-processing` for `srcNum` because the `scrNum` will always be represented by `width` bits if this method is invoked

Reply:
Thanks for your comment. The pre-processing is removed in the new commit.